### PR TITLE
Added SolveSolutionStep() to EigenSolverStrategy

### PR DESCRIPTION
**Pull request by msandre**
Following suggestion from #1911 .

### CODE REVIEW

**Comment by philbucher** [comment]
@msandre sorry for taking over this PR but I think it would be good to have all the Strategies of StructuralMechanics in one PR

@qaumann @KlausBSautter I cherry-picked your refactoring to this PR

I did all of this because I realized that  there was more to do to be consistent with the strategies in the core

Please have a look, esp @msandre and @qaumann now the Flags for Initialize and InitialieSolutionStep are used properly. 
Also you should take a close look at the FInalizeSolutionStep, imo this is not complete => you dont call FinializeSolutionStep on the Scheme and the BuilderAndSolver
I did not implement this bcs I didn't want to change the code beyond reorganizing the functions

File: applications/StructuralMechanicsApplication/custom_strategies/custom_strategies/eigensolver_strategy.hpp [hunk 1]
```diff
@@ -114,6 +104,7 @@ class EigensolverStrategy
         mpBuilderAndSolver->SetDofSetIsInitializedFlag(false);
 
         mInitializeWasPerformed = false;
+        mSolutionStepIsInitialized = false;
```
Comment (msandre) [inline applications/StructuralMechanicsApplication/custom_strategies/custom_strategies/eigensolver_strategy.hpp]: We should change these to in-class initializers in the future.

**Review by msandre — COMMENTED** [review]
@philbucher I don't think mSolutionStepIsInitialized is necessary. The base Solve() method calls this function once, then the flag is set true, then the SolveSolutionStep() is called, then the flag is set false. But the flag has no effect on the executed code. Correct me if I am wrong, but I think it just adds unnecessary complexity.

**Comment by KlausBSautter** [comment]
I did some changes on the initialized booleans in the explicit strategy -> using base class functions now

**Comment by philbucher** [comment]
@msandre in principle you are right.
I adapted what is done in the NewtonRaphson and the Linear Strategy

It is a safety measure since we can in the Future call the  functions in `Solve` separately

**Comment by qaumann** [comment]
@philbucher Thanks for merging everything into one pull request. 

I was wondering if the `InitializeSolutionStep` and `FinalizeSolutionStep` are applicable for the harmonic strategy or could be left out completely. As far as I understand, they prepare the matrices and vectors wich are passed to the linear solver in `SolveSolutionStep`. As the harmonic strategy does not need a linear solver, I just initialized all neccessary vectors as dummy vectors. Every solution step in the strategy is just a matrix vector multiplication based on the matrices and vectors initialized once in `Initialize`. 

So is it safe to remove `InitializeSolutionStep` and `FinalizeSolutionStep` from the harmonic strategy? I tried it and the tests still pass, but I am not sure if I am missing anything.

**Comment by msandre** [comment]
After talking to @philbucher I have removed the flag for InitializeSolutionStep in EigenSolverStrategy.

**Comment by philbucher** [comment]
@qaumann lets talk abt this in person, this is easier ;)

@msandre thx for removing it, I will also talk to you about one more detail but then I think it is ready for being merged :+1:

**Comment by msandre** [comment]
@philbucher I added this https://github.com/KratosMultiphysics/Kratos/pull/1978/files#diff-b9aea9229ff6a2285787bd6ce60512eeR432

**Comment by qaumann** [comment]
@philbucher as discussed, I removed the unnecessary InitializeSolutionStep and FinalizeSolutionStep form the harmonic strategy.

**Comment by philbucher** [comment]
Now everything should be fixed, could someone please rereview this?

@RiccardoRossi @loumalouomega ?

Needed for #1909

**Review by jcotela — APPROVED** [review]
Approving so we can get this into the release. I know issues have been already discussed internally.

**Comment by philbucher** [comment]
thanks

**Pull request merged**